src/test/java/space/obminyashka/items_exchange/controller/UserControllerTest.java

```diff
@@ -78,116 +81,57 @@ void negativeTestReceivingInformationAboutAnotherUser() throws Exception {
+                .concat(". ")
+                .concat(getParametrizedMessageSource("account.self.delete.request",
+                        userService.getDaysBeforeDeletion(deletedUser)))));
     }
 
     @Test
     @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn403WhenPhonesAreChanged() throws Exception {
+    void updateUserInfo_badAmountPhones_shouldReturnHttpStatusNotAcceptable() throws Exception {
         when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
-        when(userService.update(any(), any())).thenThrow(new IllegalOperationException(
-                getMessageSource("exception.illegal.field.change") + user.getPhones()));
-
-        MvcResult result = getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithPhones(), status().isForbidden())
+        MvcResult mvcResult = getResultActions(HttpMethod.PUT, "/user/info",
+                createUserUpdateDtoWithInvalidAmountOfPhones(), status().isNotAcceptable())
                 .andDo(print())
                 .andReturn();
-        assertTrue(result.getResponse().getContentAsString().contains(
-                getMessageSource("exception.illegal.field.change") + user.getPhones()));
-    }
-
-    @Test
-    @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn400WhenShortFirstName() throws Exception {
-        getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithInvalidShortFNameWithoutPhones(), status().isBadRequest())
-                .andReturn();
+        var responseAsString = mvcResult.getResponse().getContentAsString();
+        var expectedErrorMessage = getParametrizedMessageSource("exception.phones-amount", maxPhonesAmount);
+        assertTrue(responseAsString.contains(expectedErrorMessage));
     }
 
     @Test
     @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn400WhenFirstNameContainsTwoWords() throws Exception {
-        getResultActions(HttpMethod.PUT, "/user/info",
-                createUserDtoForUpdatingWithInvalidFNameWithoutPhones(), status().isBadRequest())
-                .andReturn();
-    }
+    void updateUserInfo_invalidFirstAndLastName_shouldReturnHttpStatusBadRequest() throws Exception {
+        UserUpdateDto dto = createUserUpdateDtoWithInvalidFirstAndLastName();
 
-    @Test
-    @WithMockUser(username = "admin")
-    void updateUserInfo_shouldReturn400WhenDuplicateEmail() throws Exception {
-        when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
-        when(userService.existsByEmail(any())).thenReturn(true);
+        final var errorMessageForInvalidFirstName = getMessageSource("invalid.first-or-last.name")
+                .replace("${validatedValue}", dto.getFirstName());
+        final var errorMessageForInvalidLastName = getMessageSource("invalid.first-or-last.name")
```

Comment:
U make here twice the same action. It will be better to extract it into a new method and to invoke with changable parameter.

Reply:
ok